### ISSUE TITLE
redbean: streaming

### ISSUE THREAD

**Issue by tomberek** [jart/cosmopolitan#159]
https://github.com/jart/cosmopolitan/blob/c029e83dd893b733c13b1859c9103f279a14aa37/tool/net/redbean.c#L1700 does not seem structured to allow streaming output from the Lua backend. Things like mjpg, Server-sent events, transfer-encoding: chunked, etc would need the ability to flush a body header.

Not sure if this is a desired capability.

**Comment by jart** [comment]
It is a desired capability. Along with websockets and long-polling. Right now it's the one primary limitation about redbean. It is something I intend to change in the future. 

The reason why the limitation exists, is because corking/nagle/etc. has always been controversial with TCP implementations. I wanted to make sure that whatever mechanism is provided is doing the best possible job. I feel like a feature such as this needs to be better planned out than I had time for in the first releases.

**Comment by cloudspeech** [comment]
I really would like to see Server-Sent-Events (SSE) support in redbean! 

My use case is SSE-powered chat: POST to /chat/<chatRoomID>, subscribe from n clients to receive SSE.

redbean with SSE for me would promise to be vastly simpler than nginx + nchan.io, plus easier to understand and (maybe) less buggy.

**Comment by pkulchenko** [comment]
@cloudspeech, @tomberek, [I pushed the changes that implement "flushing"](https://github.com/pkulchenko/cosmopolitan/tree/redbean-stream-yield), although it's implemented by calling `coroutine.yield()` (instead of `Flush()`). It provides a bit more flexibility (as it allows passing values back and forth if needed) and was easier to implement by piggybacking on the existing streaming implementation in Redbean.

Here is a small example:

```lua
function OnHttpRequest()
  Write("Hello, World!")
  coroutine.yield() -- this will flush the output
  Sleep(3)
  Write("Hello more")
end
```

I haven't submit a PR yet, as I had to uncomment a workaround that @jart added to protect against a memory issue (as documented in https://github.com/jart/cosmopolitan/issues/97#issuecomment-973804918).

**Comment by pkulchenko** [comment]
I made a couple of updates (in the same branch) that make the code executed by `ServeLua`  to accept `coroutine.yield()` as well and using one from `OnHttpRequest` and then calling `Route`/`RoutePath` (which may resolve to `ServeLua`) works too. For obvious reasons calling `ServeLua` after a `coroutine.yield()` is not going to reset the output (as it was already sent out).

**Comment by pkulchenko** [comment]
@tomberek, @cloudspeech, this has been implemented and merged in #370; the example shown above (https://github.com/jart/cosmopolitan/issues/159#issuecomment-1000612395) should work. Let me know if you run into any issues.
